### PR TITLE
Add quality tools

### PR DESCRIPTION
**Pull request by strangerkir**
Addresses [ECUR-58](https://inpsyde.atlassian.net/browse/ECUR-58).

* Add `phpcs` to the project, fix (or ignore sometimes) errors, add it to the GitHub actions.
* Fix `psalm` errors, add it to the GitHub actions.
* Refactor code, where needed to avoid `psalm` and `phpcs` errors.
* Update `psalm` and `phpcs` configs.
* Add `PHP 7.3` and `PHP 7.4` to the GitHub actions matrix.
* Rename `psalm` and `phpunit` config files to *.dist.
* Add functions and classes stubs for `psalm`.

All changes are related to the new code. The old code parts left completely as they are and not tested by quality tools.

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0"?>
+<ruleset name="Idealo For WooCommerce"
```
Comment (mmaymo) [inline phpcs.xml.dist:2]: Surely not so important, but do we want references to Idealo in here?

File: src/EcurrungException.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types=1);
+
+namespace Ecurring\WooEcurring;
+
+use Exception;
+
+/**
+ * Super type for all app exceptions.
+ */
+class EcurrungException extends Exception
```
Comment (mmaymo) [inline src/EcurrungException.php:12]: Maybe there is a typo in the name of this class? EcurringException?

**Review by mmaymo — COMMENTED** [review]
Just 2 minor details, the rest looks good to me

File: phpcs.xml.dist (hunk 1 above)
Comment (strangerkir) [reply phpcs.xml.dist]: Oh, my bad. Just copy & pasted the config with a few changes, but forgot to update this line.
Fixed.

File: src/EcurrungException.php (hunk 2 above)
Comment (strangerkir) [reply src/EcurrungException.php]: Yep, it's a typo, thanks. Fixed.

**Review by mmaymo — APPROVED** [review]

**Pull request merged**